### PR TITLE
Compile fixes for release config

### PR DESCRIPTION
**Pull request by rgba16f**
Mark parameters used in AZ_Error/Warning/TracePrintf as maybe_unused
ifdef out lambda's that just call those macros

Signed-off-by: rgba16f <82187279+rgba16f@users.noreply.github.com>

### CODE REVIEW

**Review by amzn-sj — APPROVED** [review]
Thanks! There's also the [Print](https://github.com/o3de/o3de/blob/eb231eaa0b1ed359f1bd262fc36999c613b9683a/Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp#L25), [Warning](https://github.com/o3de/o3de/blob/eb231eaa0b1ed359f1bd262fc36999c613b9683a/Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp#L33), and [Error](https://github.com/o3de/o3de/blob/eb231eaa0b1ed359f1bd262fc36999c613b9683a/Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp#L43) functions in [ScriptAutomationScriptBindings.cpp](https://github.com/o3de/o3de/blob/development/Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp)

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp [hunk 1]
```diff
@@ -20,34 +20,40 @@ namespace ScriptAutomation
 {
     namespace Bindings
     {
-        void Print(const AZStd::string& message)
+        void Print(const AZStd::string& message [[maybe_unused]])
         {
+#ifndef RELEASE //AZ_TracePrintf does nothing in release, ignore this call
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp:25]: The define is `_RELEASE`

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp (hunk 1 above)
Comment (rgba16f) [reply Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp:25]: It seems both `RELEASE` and `_RELEASE` are in use
`RELEASE`
https://github.com/o3de/o3de/blob/eb231eaa0b1ed359f1bd262fc36999c613b9683a/Code/Framework/AzCore/AzCore/Debug/Trace.h#L284
`_RELEASE`
https://github.com/o3de/o3de/blob/eb231eaa0b1ed359f1bd262fc36999c613b9683a/Code/Framework/AzCore/AzCore/Debug/Budget.h#L52

**Review by galibzon — APPROVED** [review]
I just learned that ` [[maybe_unused]] ` can be placed after the variable name.

File: Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp:25]: Underscore release is a lot more common than release.
It seems like we add
[RELEASE](https://sourcegraph.com/search?q=context:global+repo:github%5C.com/o3de/o3de+file:.*%5C.cpp%24+%5CbRELEASE%5Cb&patternType=regexp&case=yes)
[_RELEASE](https://sourcegraph.com/search?q=context:global+repo:github%5C.com/o3de/o3de+file:.*%5C.cpp%24+%5Cb_RELEASE%5Cb&patternType=regexp&case=yes)

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**